src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```diff
@@ -318,6 +321,209 @@ private List<Diagnostic> GetAllDiagnostics(HttpRequestParseResult parseResult)
+                    && offset != 0)
+                {
+                    dateTimeOffset = dateTimeOffset.AddOffset(offset, match.Groups["option"].Value);
+                }
+
+                text = string.Concat(text.Substring(0, match.Index), dateTimeOffset.ToUnixTimeSeconds().ToString(), text.Substring(match.Index + match.Value.Length));
+            }
+            catch (ArgumentException)
+            {
+                return node.CreateBindingFailure(HttpDiagnostics.InvalidOffset(match.Groups["offset"].Value));
+            }
+            return node.CreateBindingSuccess(text);
+        } else
+        {
+            return node.CreateBindingFailure(HttpDiagnostics.TimestampFormatError(expression));
+        }
+        
+    }
+
+    private HttpBindingResult<object?> GetDateTime(HttpExpressionNode node, string expression, MatchCollection matches, Regex pattern)
+    {
+        var text = expression;
+        var currentDateTimeOffset = DateTimeOffset.UtcNow;
+
+        // We just pre-matched the prefix, so we need to re-match the full pattern to get the type, offset, and option
+        matches = pattern.Matches(text);
+
+        string format;
+        DateTimeOffset dateTimeOffset = currentDateTimeOffset;
+        var match = matches.FirstOrDefault();
+        if (match?.Groups.Count == 4)
+        {
+            try
+            {
+                IFormatProvider formatProvider = Thread.CurrentThread.CurrentUICulture;
+
+                Group type = match.Groups["type"];
+                if (match.Groups["offset"].Success && match.Groups["option"].Success)
+                {
+                    if (int.TryParse(match.Groups["offset"].Value, out int offset) && offset != 0)
+                    {
+                        dateTimeOffset = dateTimeOffset.AddOffset(offset, match.Groups["option"].Value);
+                    }
+                }
+
+                if (string.Equals(type.Value, "rfc1123", StringComparison.OrdinalIgnoreCase))
+                {
+                    // For RFC1123, we want to be sure to use the invariant culture,
+                    // since we are potentially overriding the format for local date time
+                    // we should explicitly set the format provider to invariant culture
+                    formatProvider = CultureInfo.InvariantCulture;
+                    format = "r";
+                }
+                else if (string.Equals(type.Value, "iso8601", StringComparison.OrdinalIgnoreCase))
+                {
+                    format = "o";
+                }
+                else
+                {
+                    format = type.Value.Substring(1, type.Value.Length - 2);
+                }
+
+                text = dateTimeOffset.ToString(format, formatProvider);
+
+                if(!DateTimeOffset.TryParse(text, out dateTimeOffset) )
+                {
+                    return node.CreateBindingFailure(HttpDiagnostics.DateTimePatternMatchError(expression));
```

Comment:
Looks like we can only arrive here is the `format` is not recognized. Should we use a diagnostic message that specifically identifies that the problem is with the `format` or `type` portion of the string?